### PR TITLE
Calculation for the rate of edited task added

### LINKED ISSUE #589: Expand the statistical evaluation for course result
The goal is to create a processing rate for the tasks. It should be calculated, how many participants tried to solve the tasks. The calculated values should then be displayed in a suitable diagram.

### PR DESCRIPTION
**Pull request by Jonas960**
resolves #589

### CODE REVIEW

**Review by Zitrone44 — APPROVED** [review]
:+1:

File: web-gui/src/app/page-components/course-detail/course-results/results-statistic/results-statistic.component.ts [hunk 1]
```diff
@@ -168,4 +209,29 @@ public barChartPlugins = [];
     )
       .subscribe();
   }
+
+  showRate() { //Rate of tasks that have been edited at least once
+    this.tasks.pipe(map(t => t.map(t => t.name))).subscribe(names => this.lineChartLabels = names);
+    this.courseResults.pipe(map((extractedCResult2) => { //Calculation of the rate
+        return extractedCResult2.reduce((acc, extractedCResult) => {
+          extractedCResult.results.forEach((t) => {
+            if (acc[t.task.name] == null) acc[t.task.name] = [];
```
Comment (mxsph) [inline web-gui/src/app/page-components/course-detail/course-results/results-statistic/results-statistic.component.ts:218]:
```suggestion
            if (!acc[t.task.name]) acc[t.task.name] = [];
```

**Review by mxsph — CHANGES_REQUESTED** [review]
During testing I found a few errors:

- [ ] When opening the subtasks the diagram is moved to the right and down. It would be better if the diagram stays on the same position.
  
  error-1


- [ ] If you don't click on the bars in the diagram, you get to the subtasks, but the numbers don't change.

  error-2

- [ ] There is an error in the calculation of the statistics (see gif). Shouldn't it be that `Test` and `Test 3` have only one bar? (Since no one has yet passed the task).  

  error-3

**Comment by Jonas960** [comment]
@mxsph I have implemented your change requests. Please check if it is fine now.

**Review by mxsph — CHANGES_REQUESTED** [review]
Looks good so far :+1: , but there is still one problem:

Since you use the `name` of a task to save the calculation (and also to save which task is selected) statistics for tasks with the same name are calculated incorrectly. For this you should use the task ID, as this is unique.

**Comment by Jonas960** [comment]
grafik
grafik

@mxsph I fixed it.

**Comment by mxsph** [comment]
> @mxsph I fixed it.

The calculation looks good now :+1:

However, there is still the same problem with the display of subtasks since you use there also only the name.

**Comment by Jonas960** [comment]
@mxsph The presentation of the subtasks is now based on the ids of the parent tasks. I tested it with some tasks and it worked for me.

**Review by mxsph — APPROVED** [review]

**Pull request merged**